### PR TITLE
Account for associated consts in the "unstable assoc item name colission" lint

### LINKED ISSUE #81663: unstable_name_collisions lint gives wrong message and suggestion for associated constants.
The `unstable_name_collisions` lint calls an associated constant a 'method', and suggests `Trait::CONST(...)` instead of `<... as Trait>::CONST`.

Example:

```
warning: a method with this name may be added to the standard library in the future
    --> src/atof/algorithm/math.rs:1909:34
     |
1909 | const ALGORITHM_D_B: Wide = 1 << Limb::BITS;
     |                                  ^^^^^^^^^^
     |
     = note: `#[warn(unstable_name_collisions)]` on by default
     = warning: once this method is added to the standard library, the ambiguity may cause an error or change in behavior!
     = note: for more information, see issue #48919 <https://github.com/rust-lang/rust/issues/48919>
     = help: call with fully qualified syntax `util::num::Integer::BITS(...)` to keep using the current method
```

### PR DESCRIPTION
**Pull request by estebank**
Fix #81663.

### CODE REVIEW

File: compiler/rustc_typeck/src/check/method/probe.rs [hunk 1]
```diff
@@ -1246,24 +1247,46 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
...
+                let def_kind = stable_pick.item.kind.as_def_kind();
+                let mut diag = lint.build(&format!(
+                    "{} {} with this name may be added to the standard library in the future",
+                    def_kind.article(),
+                    def_kind.descr(stable_pick.item.def_id),
+                ));
                 // FIXME: This should be a `span_suggestion` instead of `help`
                 // However `self.span` only
                 // highlights the method name, so we can't use it. Also consider reusing the code from
                 // `report_method_error()`.
```
Comment (oli-obk) [inline compiler/rustc_typeck/src/check/method/probe.rs:1266]: I guess this FIXME is now half-resolved? can you update it or move it into the first match arm?

**Comment by oli-obk** [comment]
r=me with nit resolved

File: compiler/rustc_typeck/src/check/method/probe.rs (hunk 1 above)
Comment (estebank) [reply compiler/rustc_typeck/src/check/method/probe.rs:1266]: My interpretation of the FIXME is that we still have the same problem for method calls. We are only handling the assoc const case correctly now (which we didn't handle at all before). I can move it to the appropriate arm though.

**Pull request merged**